addHack.js

```diff
@@ -19,13 +19,23 @@ export const add = async (event, context) => {
     });
     conn.model(
       "Hack",
-      new mongoose.Schema({ title: String, description: String, goal: String, team: Array })
+      new mongoose.Schema({
+        title: { type: String, default: "" },
+        description: { type: String, default: "" },
+        goal: { type: String, default: "" },
+        team: { type: Array, default: [] }
+      })
     );
   }
   const Hack = conn.model("Hack");
 
+  if (title) newIdea.title = title;
+  if (description) newIdea.description = description;
+  if (goal) newIdea.goal = goal;
+  if (team) newIdea.team = team;
```

Comment:
I feel like I've never needed to do this before, and feels a bit heavy handed, wondering if there is another way it could be done.

My initial instinct is a little utility function to just make it more scalable... something like.

```
const pickIfTruthy = (data, ...fields) => fields.reduce((newObject, field) => {
  if(data[field]) {
    newObject[field] = data[field]
  }
  return newObject
}, {})
```

Then you can do 

```
const newIdea = pickIfTruthy(data, 'idea', 'description', 'goal', 'team').
```

Reply:
it's awesome to see reduce in action, let me try this out!

Reply:
I should also mention that there are options from lodash that can do some of this depending on what you want to do... `pickBy` or `defaults` would be the ones I'd look at.